app/models/host/managed.rb

```diff
@@ -239,6 +239,20 @@ def built(installed = true)
     end
   end
 
+  # rebuilds TFTP, DHCP, DNS for the host
+  def rebuild_config
+    result = {:tftp => false, :dhcp => false, :dns => false}
+    nic = interfaces.select(&:provision).first
+    begin
+      result[:tftp] = nic.setTFTP if nic.tftp?
+    rescue
```

Comment:
Where you rescue exceptions isn't consistent - DNS and DHCP are both rescued inside the rebuild methods, TFTP is rescued here.  I'm not sure if they should be rescued or not though, see above.  If they should be, I'd create a rebuild_tftp method and do it there for consistency.